docs/DeveloperGuide.md

```diff
@@ -2,32 +2,143 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+- NUS SC2113 Duke project
+- [NUS SC2113 tp](https://github.com/nus-cs2113-AY2223S2)
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+All commands will be fetched by the getUserCommand() method in the TextUi class. The parseCommand() method in the Parser class will then parse the command and pass it to the respective command classes. The command classes will be responsible for executing the relevant features.
+
+### [Proposed] Search book by title
+
+![](images/Search%20class%20diagram.png)
+Description: Searches for a book using the title  
+Format: search -title TITLE
+Example:
+- search -title Python Programming  
+For successful searches, the program will output the relevant book along with the book details: ISBN, Title, Author, Topic. For unsuccessful searches, the program will output a string informing the user that there is no match with the input title from the inventory.
+
+Note: Partial string matching will not be considered in this application. Exact match of title and topic will be done, and if match is found, the book object will be returned.
+
+### [Proposed]  Search book by topic
+Description: Searches for a book by its topic  
+Format: search -topic TOPIC
+Example:
+- search -topic Business  
+For successful searches, the program will output the relevant books along with the book details: ISBN, Title, Author, Topic. For unsuccessful searches, the program will output a string informing the user that there is no match with the input topic from the inventory.
+
+
+### [Proposed]  Borrow book
+Description: Borrow a book from the library  
+Format: borrow -title TITLE
+Example:
+- borrow -title Python Programming  
+For successful borrowing, the program will output a string showing that the action is successful and, at the same time, mark the book as borrowed in the system. For unsuccessful borrow requests, the program will either output that there is no such book in the inventory or a message showing that the book is already on loan at the time of the borrow request.
```

Comment:
Perhaps you could include more technical documentation on how the command runs algorithmically.

Same goes for the rest of the features! That way it will feel more like a DG instead of a UG.